add a new proposal to sig-robotics: Offline Visual Navigation without Metric Map

chch9907 opened the pull request:
This is the project of my ICRA paper: SCALE: Self-Correcting Visual Navigation for Mobile Robots via
Anti-Novelty Estimation

chch9907 commented:
/assign @hongbingzhang

fujitatomoya reviewed (commented):
@chch9907 thanks for the proposal.

Although this is really nice framework for robotics, I would like to have a few questions.

- This seems pretty much system service application, not implemented in KubeEdge. So the proposal is to develop the system application containers, and deploy it on KubeEdge if user wants to do so?
- If that is so, how user call these APIs or interfaces? I guess that image sensing data comes from edge devices where edgecore runs, and some pre-process can be done on edge. but expecting some processes need to be running in the cloud infrastructure, in that case how we can connect those data pipeline at runtime? Or is this supposed to be a single pod to process everything?

I may be missing something, so would like to hear feedback to learn more.

chch9907 commented on sig-robotics/proposal/Offline Visual Navigation without Metric Map/Offline Visual Navigation without Metric Map.md:
```diff
@@ -0,0 +1,72 @@
+
+## Proposal
+
+### Project scope
```
Do you mean deleting line 19 or deleting the unnecessary space in line 20? I am not familar with it.

fujitatomoya replied:
No i mean adding the blank line between section title and description.

chch9907 commented on sig-robotics/proposal/Offline Visual Navigation without Metric Map/Offline Visual Navigation without Metric Map.md:
```diff
@@ -0,0 +1,72 @@
+# Offline Visual Navigation without Metric Map
+
+## Background
```
Do you mean deleting line 3 or deleting the unnecessary space in line 4? I am not familar with it.

fujitatomoya replied:
same here.

chch9907 commented on sig-robotics/proposal/Offline Visual Navigation without Metric Map/Offline Visual Navigation without Metric Map.md:
```diff
@@ -0,0 +1,72 @@
+# Offline Visual Navigation without Metric Map
+
+## Background
+Existing learning-based navigation studies mainly develop the algorithms in simulation environment that often suffer from **sim-to-real gap** issue when transferring to the real world. Recently, there is some work dedicated to lift the simulation assumptions by learning directly from offline real-world dataset that are not necessarily collected from experts, such as ViNG [1], ReViND [2], and ViNT ...
```
Do you mean deleting the whole first paragraph?

fujitatomoya replied:
git does not pick the diff here, but i added `has` in `This route has been proven`. nit...

chch9907 commented:
> @chch9907 thanks for the proposal.
> 
> Although this is really nice framework for robotics, I would like to have a few questions.
> 
> * This seems pretty much system service application, not implemented in KubeEdge. So the proposal is to develop the system application containers, and deploy it on KubeEdge if user wants to do so?
> * If that is so, how user call these APIs or interfaces? I guess that image sensing data comes from edge devices where edgecore runs, and some pre-process can be done on edge. but expecting some processes need to be running in the cloud infrastructure, in that case how we can connect those data pipeline at runtime? Or is this supposed to be a single pod to process everything?
> 
> I may be missing something, so would like to hear feedback to learn more.

Thank you for your comments and useful code review. This proposal is indeed not implemented in KubeEdge but to develop a system application. As this is an offline-learned navigation method without sim-to-real gaps, it requires the real-world offline trajectory datasets that are absent in current open-sourced datasets, particulary those accrosing different illuminations and seasons. Therefore, this proposal aims to leverage KubeEdge to curate the real-world datasets collected from diverse edge devices for training the navigation methods that are generalized to different scenarios and robust to different illuminations or seasons . In return, the end users can then use the trained models from the cloud to conduct the visual navigation and localization in their deployed environments with a image-node topological map representation. This is a promising route for handling real-world visual navigation tasks. Furthermore, this model is not too large to access from cloud at runtime, and can even be directly customized on edge, which has been tested during our experiments.  

I will illustrate this idea in the proposal. Thank you for your suggestions!

fujitatomoya commented on sig-robotics/proposal/Offline Visual Navigation without Metric Map/Offline Visual Navigation without Metric Map.md:
```diff
@@ -0,0 +1,72 @@
+  
+
+
+### Data Collection Module
```
same with else where.

fujitatomoya commented:
@chch9907 thanks for the explanation, since this is proposal with concrete algorithm, i am good to go with this. I would like to other reviews to take a look before merge.

chch9907 commented:
I get lost in using PR..

chch9907 commented:
/assign @hongbingzhang

chch9907 commented:
@fisherxu Hello, could you help review my PR? I have improved the proposal according to the comments.

chch9907 commented:
@fisherxu Hi xu, I have deleted the others' commits that confused me before, and squashed my commits into one. But it seems there is something wrong in my operation. Could you help me? If no problem, could you help merge it? Thanks!
